### PR TITLE
 Adding possibility to get one random deck.

### PR DESCRIPTION
**Pull request by ArekArek**
Added in DeckDao option to get random deck from database,
    Added in DecksResource optional path param which is responsible for getting one random deck,
    Added test accordingly.

### CODE REVIEW

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 1]
```diff
@@ -42,30 +42,39 @@ public Deck createDeck(@Auth @Valid DeckRepresentation deck, @Context SecurityCo
...
                 else {
                     decks.addAll(decksDAO.getAllUserDecksWithFlashcardsNumber(user.getId()));
                     return decks;
                 }
             }
-        }
-        else {
-            Collection<Deck> decks = decksDAO.getDecksByName(name, user.getId());
+        } else {
+            decks = decksDAO.getDecksByName(name, user.getId());
```
Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: This resource is becoming very complicated. We should figure out some other way to parse query params, e.x. Builder or some other way

**Comment by VanDavv** [comment]
PR reviewed. It has some issues, but they're not crucial for this PR, and will be fixed soon

File: backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java [hunk 2]
```diff
@@ -42,30 +42,39 @@ public Deck createDeck(@Auth @Valid DeckRepresentation deck, @Context SecurityCo
...
 
-                if(includeOwn == null || !includeOwn) return decks;
-                else {
+                decks = decksDAO.getAllDecks(user.getId());
+
+                if (includeOwn == null || !includeOwn) {
+                    if (random == null || !random) {
+                        return decks;
+                    } else {
+                        return decksDAO.getRandomDecks(user.getId());
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksResource.java]: In line 53 you pulled all decks for user. Wy don't you pick some random decks from 'decks'?

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**